### PR TITLE
Add detect-changes action

### PR DESCRIPTION
**Pull request by OlegKipchatov**

### CODE REVIEW

File: detect-changes/action.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+name: Detect Geteneration Changes
```
Comment (krutilin) [inline detect-changes/action.yml]: Missprint.

File: detect-changes/action.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+inputs:
+  path:
+    description: Path to directory or file to check for changes
+    required: true
+
+outputs:
+  changed:
+    description: 'true' if changes in the generator are detected
```
Comment (IlyaKhD) [inline detect-changes/action.yml]:
```suggestion
    description: 'true' if changes detected
```

Comment (IlyaKhD) [reply detect-changes/action.yml]: This can be used not only for generator files, right?

File: detect-changes/action.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+  steps:
+    - shell: bash
+      id: detection
+      run: |
+        diff=$( git diff HEAD^ --name-only -- ${{ github.event.inputs.path }} )
+
+        if [ "$diff" == "" ]
+        then
+          echo "::set-output name=changed::false"
+          echo "The generator files have no changes"
```
Comment (IlyaKhD) [inline detect-changes/action.yml]:
```suggestion
          echo "No changes detected"
```

**Review by IlyaKhD — APPROVED** [review]

**Review by krutilin — APPROVED** [review]

**Review by alexslavr — APPROVED** [review]

**Pull request closed without merging**